localstack-core/localstack/services/apigateway/legacy/provider.py

```diff
@@ -2392,6 +2402,9 @@ def get_api_keys(
         if not include_values:
             for api_key in api_keys:
                 api_key.pop("value")
+        if limit is not None:
+            if limit < 1 or limit > 500:
+                limit = None
```

Comment:
nit: I think the PR description mentions pagination, but there are no tests for it, so I'd suggest to just drop this new code and merge it like that, the newly added validation is already really good 👍